pkg/defaults/defaults.go

```diff
@@ -384,6 +384,9 @@ const (
 	// pre-allocations
 	IPAMMultiPoolPreAllocation = "default=8"
```

Comment:
Hm, so this is something I did not consider - it seems that we're still using `default` for the `ipam-multi-pool-pre-allocation` flag, which means that if a user only sets `ipam-default-ip-pool`, the local node will still attempt to pre-allocate from the `default` pool.

I would propose the following:
1. Let's set the default value of `ipam-multi-pool-pre-allocation` to the empty string [here](https://github.com/cilium/cilium/blob/c0c422d7389d78e3617b56a4efc40aee329c4833/daemon/cmd/daemon_main.go#L301)
2. In `option.Config.Populate()`, after we've parsed the flag, let's add a check that if `c.IPAMMultiPoolPreAllocation` is an empty map that we set it to `${c.IPAMDefaultIPPool}=8 [here](https://github.com/cilium/cilium/blob/d6b1637880da13ab76c8465851c08c630e93f10c/pkg/option/config.go#L3555)

That way, the default pool will be pre-allocated even if it's not named "default"

Reply:
Got it.